File: CS/Q253018/BarHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars;

namespace Q253018
{
    public class BarHelper
    {
        BarManager manager;

        public BarHelper(BarManager manager)
        {
            this.manager = manager;
        }


        public bool ExpandMenuOnMouseHover
        {
            set
            {
                manager.HighlightedLinkChanged -= OnBarManagerHighlightedLinkChanged;
                if (value)
                    manager.HighlightedLinkChanged += OnBarManagerHighlightedLinkChanged;
            }
        }

        private void OnBarManagerHighlightedLinkChanged(object sender, HighlightedLinkChangedEventArgs e)
        {
            Point mousePosition = manager.Form.PointToClient(Control.MousePosition);
            BarCustomContainerItemLink link = e.Link as BarCustomContainerItemLink;
            if (link != null && link.Bounds.Contains(mousePosition))
                link.OpenMenu();
        }

    }
}

File: VB/Q253018/BarHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars

Namespace Q253018
	Public Class BarHelper
		Private manager As BarManager

		Public Sub New(ByVal manager As BarManager)
			Me.manager = manager
		End Sub


		Public WriteOnly Property ExpandMenuOnMouseHover() As Boolean
			Set(ByVal value As Boolean)
				RemoveHandler manager.HighlightedLinkChanged, AddressOf OnBarManagerHighlightedLinkChanged
				If value Then
					AddHandler manager.HighlightedLinkChanged, AddressOf OnBarManagerHighlightedLinkChanged
				End If
			End Set
		End Property

		Private Sub OnBarManagerHighlightedLinkChanged(ByVal sender As Object, ByVal e As HighlightedLinkChangedEventArgs)
			Dim mousePosition As Point = manager.Form.PointToClient(Control.MousePosition)
			Dim link As BarCustomContainerItemLink = TryCast(e.Link, BarCustomContainerItemLink)
			If link IsNot Nothing AndAlso link.Bounds.Contains(mousePosition) Then
				link.OpenMenu()
			End If
		End Sub

	End Class
End Namespace
